contrib/lang/typescript/packages.el

```diff
@@ -0,0 +1,35 @@
+;;; packages.el --- typescript Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2014 Sylvain Benner
+;; Copyright (c) 2014-2015 Chris Bowdon & Contributors
+;;
+;; Author: Chris Bowdon <c.bowdon@bath.edu>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defvar typescript-packages '(tss)
+  "List of all packages to install and/or initialize. Built-in packages
+which require an initialization must be listed explicitly in the list.")
+
+(defvar typescript-excluded-packages '()
+  "List of packages to exclude.")
+
+(defun typescript/init-tss ()
+  "Initialize my package"
+  (use-package tss
+    :mode ("\\.ts\\'" . typescript-mode)
+    :defer t
+    :init
+    (progn
+      (require 'typescript)
+      (require 'tss)
```

Comment:
these shouldn't need to be required here, that should be set up by `user-package`. If you're running into trouble with undefined names, put that code under `:config`.